### PR TITLE
HPCC-14720 Use __pause for 32bit GNU build

### PR DESCRIPTION
**Pull request by mckellyln**
Signed-off-by: Mark Kelly mark.kelly@lexisnexis.com

### CODE REVIEW

**Comment by mckellyln** [comment]
@ghalliday @xwang2713 please review.

File: system/jlib/jatomic.hpp [hunk 1]
```diff
@@ -23,8 +23,15 @@
 #ifdef _WIN32
 inline static void spinPause() { YieldProcessor(); }
 #else
-#include "xmmintrin.h"
+# include "x86intrin.h"
+# if defined(_ARCH_X86_)
```
Comment (ghalliday) [inline system/jlib/jatomic.hpp]: Should the #include be inside the #if?
Are there problems with building on non x86 machines if we include that header?
Does it also work on apple?

Comment (richardkchapman) [inline system/jlib/jatomic.hpp]: I think it is compiling on OSX (though 6.0.0 branch fails to build on OSX for other reasons at the moment)

Comment (richardkchapman) [inline system/jlib/jatomic.hpp]: I don't believe the include should be within the if, though we may need a separate outer if to handle non-intel architectures.

Comment (mckellyln) [inline system/jlib/jatomic.hpp]: We dont seem to have a generic intel/amd/x86* macro for the platform to put the if within, but we could do this -

\\#if defined(_ARCH_X86_64_) || defined(_ARCH_X86_)
  \\#include "x86intrin.h"
  \\#if defined(_ARCH_X86_)
  inline static void spinPause() { __pause(); }
  \\#else
  inline static void spinPause() { _mm_pause(); }
  \\#endif
\\#endif

**Comment by mckellyln** [comment]
updated to only include x86intrin include file for x86_64 or x86.

**Comment by richardkchapman** [comment]
@ghalliday Please re-review

**Pull request merged**